src/lib.rs

```diff
@@ -4,52 +4,57 @@
 extern crate delog;
 generate_macros!();
 
-pub use heapless_bytes::Bytes;
-
 pub(crate) mod consts;
 pub mod de;
 pub mod error;
 pub mod ser;
 
 pub use error::{Error, Result};
+use ser::Writer;
 
-// pub use de::from_bytes;
-// pub use de::take_from_bytes;
+/// Serialize an object to a `Writer`
+///
+/// Returns the amount of bytes written to the writer
+pub fn cbor_serialize_to<'a, T: ?Sized + serde::Serialize, W: Writer>(
+    object: &T,
+    writer: W,
+) -> Result<usize> {
+    let mut serializer = ser::Serializer::new(writer);
+    object.serialize(&mut serializer)?;
+    Ok(serializer.written())
+}
 
 // kudos to postcard, this is much nicer than returning size
 pub fn cbor_serialize<'a, T: ?Sized + serde::Serialize>(
     object: &T,
     buffer: &'a mut [u8],
 ) -> Result<&'a [u8]> {
-    let writer = ser::SliceWriter::new(buffer);
-    let mut ser = ser::Serializer::new(writer);
-
-    object.serialize(&mut ser)?;
-
-    let writer = ser.into_inner();
-    let size = writer.bytes_written();
-
-    Ok(&buffer[..size])
+    let mut buf = &mut *buffer;
+    let written = cbor_serialize_to(object, &mut buf)?;
+    Ok(&buffer[..written])
 }
 
+#[cfg(feature = "heapless-bytes-v0-3")]
 /// Append serialization of object to existing bytes, returning length of serialized object.
 pub fn cbor_serialize_extending_bytes<T: ?Sized + serde::Serialize, const N: usize>(
     object: &T,
-    bytes: &mut Bytes<N>,
+    bytes: &mut heapless_bytes_v0_3::Bytes<N>,
 ) -> Result<usize> {
     let len_before = bytes.len();
-    let mut ser = ser::Serializer::new(bytes);
+    let vec: &mut heapless_v0_7::Vec<u8, N> = bytes;
+    let mut ser = ser::Serializer::new(vec);
 
     object.serialize(&mut ser)?;
 
     Ok(ser.into_inner().len() - len_before)
 }
 
+#[cfg(feature = "heapless-bytes-v0-3")]
 /// Serialize object into newly allocated Bytes.
 pub fn cbor_serialize_bytes<T: ?Sized + serde::Serialize, const N: usize>(
```

Comment:
Same as above – a replacement could use `Writer + Default`.

Reply:
Same as above, kept for backwards compatibility. But maybe we tag this as a nitrokey release, and merge it removing the old function (maybe in a separate PR, that cleans everything up for 0.5.0).